js/controller.js

```diff
@@ -1299,6 +1324,24 @@ OO.plugin('Html5Skin', function(OO, _, $, W) {
       // In case ad was skipped or errored while stalled
       this.setBufferingState(false);
       this.renderSkin();
+
+      // restore video360 pos by deviceOrientation changes (only for mobile devices)
+      // It's uses after video 360 was moved by device motion, after ads playback
+      if(this.state.deviceOrientation.allow) {
+        var orientationType = Utils.getOrientationType();
+        var params = [0, 0, 0];
+
+        params[0] = this.state.deviceOrientation.alpha;
+        params[2] = orientationType === 'landscape-primary' ? 
+          this.state.deviceOrientation.beta : 
+          this.state.deviceOrientation.gamma;
+
+        // To apply there need a time gap
+        setTimeout(function() {
+          this.onTouchMove(params);
```

Comment:
Could we separate this logic from the `touchmove` event? In general I think it's better not to call event handlers manually since some other part of the code might be expecting these to be triggered by actual DOM event